### PR TITLE
MC-36425: “Select Shipping Method” page of Multishipping Checkout is corrupted if a Customer use “go back” browser button to return to the page

### LINKED ISSUE #30268: “Select Shipping Method” page of Multishipping Checkout is corrupted if a Customer use “go back” browser button to return to the page
**Preconditions:**
* Magento EE 2.3.5-p2, EE 2.4.0, EE 2.4.0-p1, EE 2.4.1-alpha3{color} is installed, should be actule for CE
* A Simple product is created
* A Customer account with two shipping addresses is created
* Check / Money order Payment method is configured
* Flat Rate shipping method is configured

**Steps to Reproduce:**
* Go to the Storefront as the created Customer
* Add the created Product (2pcs) to the Shopping Cart
* Open the mini-cart
* Click the {color:#0747a6}“View and Edit Cart”{color} link
* Click the “{color:#0747a6}Check Out with Multiple Addresses”{color} link
* Select shipping addresses
* Click the “Go to Shipping Information” button
* Select Flat Rate shipping method
* Click the “Continue to Billing Information” button
* Click the "Add or remove gift cards" link on the “Billing Information” page
* Wait until the “Shopping Cart” page is loaded completely
* Click the “go back” browser button (left arrow)
* Click the {color:#0747a6}”Back to Shipping Information” link on the “Billing Information” page

**Expected results:**
* “Select Shipping Method” page should be loaded
* The page should not be corrupted

**Actual results:**

(/) “Select Shipping Method” page is loaded
 (x) The page is corrupted 

image

### PR DESCRIPTION
**Pull request by engcom-Kilo**
### Fixed Issues (if relevant)

1. Fixes https://github.com/magento/magento2/issues/30268

### Manual testing scenarios (*)

1. see https://github.com/magento/magento2/issues/30268

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Comment by engcom-Kilo** [comment]
@magento run all tests

File: app/code/Magento/Multishipping/Test/Mftf/Test/StorefrontCreateOrderWithMultishippingAfterReturningToCartTest.xml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            <!--Set up configuration.-->
+            <magentoCLI command="config:set {{EnableFreeShippingMethod.path}} {{EnableFreeShippingMethod.value}}" stepKey="enableFreeShipping"/>
+            <magentoCLI command="config:set {{EnableFlatRateShippingMethod.path}} {{EnableFlatRateShippingMethod.value}}" stepKey="enableFlatRateShipping"/>
+            <magentoCLI command="config:set {{EnableCheckMoneyOrderPaymentMethod.path}} {{EnableCheckMoneyOrderPaymentMethod.value}}" stepKey="enableCheckMoneyOrderPaymentMethod"/>
+        </before>
+        <after>
+            <!--Clean up test data, revert configuration.-->
+            <deleteData createDataKey="product" stepKey="deleteProduct"/>
+            <deleteData createDataKey="customer" stepKey="deleteCustomer"/>
+            <magentoCLI command="config:set {{DisableFreeShippingMethod.path}} {{DisableFreeShippingMethod.value}}" stepKey="disableFreeShipping"/>
```
Comment (rogyar) [inline app/code/Magento/Multishipping/Test/Mftf/Test/StorefrontCreateOrderWithMultishippingAfterReturningToCartTest.xml]: That's so awesome that we don't clean cache anymore after setting a config via CLI anymore (it's invalidated automatically in this case). 

Great job!

**Comment by engcom-Kilo** [comment]
@magento run all tests

File: app/code/Magento/Multishipping/Test/Mftf/Test/StorefrontCreateOrderWithMultishippingAfterReturningToCartTest.xml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
+    <test name="StorefrontCreateOrderWithMultishippingAfterReturningToCartTest">
+        <annotations>
+            <features value="Multishipping"/>
+            <stories value="Checkout with multiple addresses."/>
+            <title value="Checkout with multiple addresses after returning on cart page during checkout."/>
+            <description value="Verify customer able to checkout with multiple addresses after returning to cart page and continue checkout with browser 'back' button."/>
+            <severity value="AVERAGE"/>
+            <testCaseId value="MC-39583"/>
```
Comment (zakdma) [inline app/code/Magento/Multishipping/Test/Mftf/Test/StorefrontCreateOrderWithMultishippingAfterReturningToCartTest.xml:18]: Please add MC-39583 as **has tests** link to task
Please specify test Path in MC-39583 zephyr
Please add **Expected result** for steps in MC-39583 zephyr

File: app/code/Magento/Multishipping/Test/Mftf/Test/StorefrontCreateOrderWithMultishippingAfterReturningToCartTest.xml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        </before>
+        <after>
+            <!--Clean up test data, revert configuration.-->
+            <deleteData createDataKey="product" stepKey="deleteProduct"/>
+            <deleteData createDataKey="customer" stepKey="deleteCustomer"/>
+            <magentoCLI command="config:set {{DisableFreeShippingMethod.path}} {{DisableFreeShippingMethod.value}}" stepKey="disableFreeShipping"/>
+        </after>
+
+        <!--Add product to cart and proceed to multishipping checkout. -->
+        <actionGroup ref="LoginToStorefrontActionGroup" stepKey="loginToStorefrontAccount">
```
Comment (zakdma) [inline app/code/Magento/Multishipping/Test/Mftf/Test/StorefrontCreateOrderWithMultishippingAfterReturningToCartTest.xml:42]: Please add logout in **after** section but before customer delete

File: app/code/Magento/Multishipping/Test/Mftf/Test/StorefrontCreateOrderWithMultishippingAfterReturningToCartTest.xml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+            <argument name="Customer" value="$$customer$$"/>
```
Comment (zakdma) [inline app/code/Magento/Multishipping/Test/Mftf/Test/StorefrontCreateOrderWithMultishippingAfterReturningToCartTest.xml:43]: Please dont use $$. Use single $ instead

**Review by zakdma — CHANGES_REQUESTED** [review]

**Comment by zakdma** [comment]
@magento import code to https://github.com/magento-tsg/magento2ce

**Comment by magento-engcom-team** [comment]
@zakdma the branch with code successfully imported into`magento-tsg/magento2ce` repository. Branch name: `imported-magento-magento2-31156`.

**Comment by zakdma** [comment]
@magento run all tests

**Review by zakdma — APPROVED** [review]

**Comment by zakdma** [comment]
@magento run all tests

**Comment by engcom-Bravo** [comment]
:heavy_check_mark: QA Passed

### Manual testing scenario 
1. Go to the Storefront as the created Customer
2. Add the created Product (2pcs) to the Shopping Cart
3. Open the mini-cart
4. Click the {color:#0747a6}“View and Edit Cart”{color} link
5. Click the “{color:#0747a6}Check Out with Multiple Addresses”{color} link
6. Select shipping addresses
7. Click the “Go to Shipping Information” button
8. Select Flat Rate shipping method
9. Click the “Continue to Billing Information” button
10. Click the "Add or remove gift cards" link on the “Billing Information” page
11. Wait until the “Shopping Cart” page is loaded completely
12. Click the “go back” browser button (left arrow)
13. Click the {color:#0747a6}”Back to Shipping Information” link on the “Billing Information” page

**BEFORE** applying changes provided in the PR, the web page was not fully loaded, and an Exception appeared
pr_before

**AFTER** switching to the PR, the page is fully loaded with no errors 
pr_after

The order can be placed successfully

**Comment by zakdma** [comment]
@magento run Functional Tests EE, Integration Tests

**Comment by zakdma** [comment]
@magento run Functional Tests B2B

**Comment by zakdma** [comment]
@magento run Functional Tests B2B, Functional Tests EE, Integration Tests

**Comment by zakdma** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Comment by zakdma** [comment]
@magento import code to https://github.com/magento-tsg/magento2ce

**Comment by magento-engcom-team** [comment]
@zakdma the branch with code successfully imported into`magento-tsg/magento2ce` repository. Branch name: `imported-magento-magento2-31156`.

**Comment by zakdma** [comment]
@magento run all tests

**Comment by zakdma** [comment]
@magento run Functional Tests CE

**Comment by zakdma** [comment]
Changes was delivered in scope of internal ticket MC-36425

**Pull request closed without merging**